src/table/types.js

```diff
@@ -28,6 +28,8 @@ export type HeadCellPropsT = {|
   },
   /** Column title. */
   title: string,
+  /** Extend click enable click to sort on whitespace in a header cell. */
+  extendClick?: boolean,
```

Comment:
let's call this `fillClickTarget` to convey that it fills the available negative space rather than extends to an arbitrary point.